examples/direct_fidelity_estimation.ipynb

```diff
@@ -0,0 +1,228 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Running the Direct Fidelity Estimation (DFE) algorithm\n",
+    "This example walks through the steps of running the direct fidelity estimation (DFE) algorithm as described in these two paper:\n",
+    "\n",
+    "* Direct Fidelity Estimation from Few Pauli Measurements (https://arxiv.org/abs/1104.4695)\n",
+    "* Practical characterization of quantum devices without tomography (https://arxiv.org/abs/1104.3835)\n",
+    "\n",
+    "The code is optimized for Clifford circuits using tableaux\n",
+    "* Improved Simulation of Stabilizer Circuits (https://arxiv.org/pdf/quant-ph/0406196.pdf)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "0: ───@───Z───────\n",
+      "      │\n",
+      "1: ───┼───────X───\n",
+      "      │       │\n",
+      "2: ───X───H───@───\n"
+     ]
+    }
+   ],
+   "source": [
+    "# Import Cirq, DFE, and create a circuit\n",
+    "\n",
+    "import cirq\n",
+    "import direct_fidelity_estimation as dfe\n",
+    "\n",
+    "qubits = cirq.LineQubit.range(3)\n",
+    "circuit = cirq.Circuit(cirq.CNOT(qubits[0], qubits[2]),\n",
+    "                       cirq.Z(qubits[0]),\n",
+    "                       cirq.H(qubits[2]),\n",
+    "                       cirq.CNOT(qubits[2], qubits[1]))\n",
+    "\n",
+    "print(circuit)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# We then create a sampler. For this example, we use a simulator but the code can accept a hardware sampler.\n",
+    "noise = cirq.ConstantQubitNoiseModel(cirq.depolarize(0.1))\n",
+    "sampler = cirq.DensityMatrixSimulator(noise=noise)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Estimated fidelity: 0.53\n"
+     ]
+    }
+   ],
+   "source": [
+    "# We run the DFE:\n",
+    "estimated_fidelity, intermediate_results = dfe.direct_fidelity_estimation(\n",
+    "        circuit,\n",
+    "        qubits,\n",
+    "        sampler,\n",
+    "        n_measured_operators=None,\n",
+    "        samples_per_term=0)\n",
```

Comment:
You should probably indicate what the sentinel values of `None` and `0` mean in this instance (the other arguments are self-explanatory)

edit: I see later you dive into this deeper. Maybe a short comment / text saying that you'll describe these parameters later in the notebook

Reply:
Done.